libpromises/cf3parse_logic.h

```diff
@@ -1028,4 +1030,58 @@ static inline void ParserBeginBundleBody()
     P.useargs = NULL;
 }
 
+static inline void ParserHandleQuotedListItems()
+{
+    ParserDebug("\tP:%s:%s:%s:%s list append: "
+                "qstring = %s\n",
+                ParserBlockString(P.block), P.blocktype, P.blockid,
+                (P.currentclasses ?
+                    P.currentclasses : "any"),
+                P.currentstring);
+
+    if(StringContains(P.lval, "list") && *P.lval != 's')  // deal with numbers in ilist/rlist
```

Comment:
This is oddly specific and imprecise at the same time.

You wanted to capture `(ilist|rlist)`, instead you made something that matches `[^s].*list.*`

I think you should try to focus more on correctness and readability here, rather than performance and cleverness.

For example:

```C
if (StringEquals(P.lval, "ilist") || StringEquals(P.lval, "rlist"))
```

Or:

```C
if ((P.lval[0] == 'i' || P.lval[0] == 'r') && StringEquals(P.lval + 1, "list"))
```